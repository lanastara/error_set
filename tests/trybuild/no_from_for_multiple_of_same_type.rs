use error_set::error_set;

error_set! {
    X = {
        IoError(std::io::Error),
        IoError2(std::io::Error),
    };
    Y = {
        IoError2(std::io::Error),
        IoError(std::io::Error),
    };
    Z = {
        IoError(std::io::Error),
    };
}

fn main() {
    let io_error = std::io::Error::new(std::io::ErrorKind::OutOfMemory, "oops out of memory");
    let z: Z = io_error.into();
    let x: X = z.into();
    matches!(x, X::IoError(_));
    let y: Y = x.into();
    matches!(y, Y::IoError2(_));
    let x: X = y.into();
    matches!(x, X::IoError(_));
    let io_error = std::io::Error::new(std::io::ErrorKind::OutOfMemory, "oops out of memory");
    let y: Y = io_error.into();
    matches!(y, Y::IoError2(_));
}