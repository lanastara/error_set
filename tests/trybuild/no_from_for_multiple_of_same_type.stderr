error[E0277]: the trait bound `Y: From<std::io::Error>` is not satisfied
  --> tests/trybuild/no_from_for_multiple_of_same_type.rs:27:25
   |
27 |     let y: Y = io_error.into();
   |                         ^^^^ the trait `From<std::io::Error>` is not implemented for `Y`, which is required by `std::io::Error: Into<_>`
   |
   = help: the following other types implement trait `From<T>`:
             `Y` implements `From<X>`
             `Y` implements `From<Z>`
   = note: required for `std::io::Error` to implement `Into<Y>`
